src/keysValueValidator.js

```diff
@@ -0,0 +1,63 @@
+
+import { keyTypes } from './constants.js';
+import { createError } from './error.js';
+import {maxLength,isEncoded,validNrrFormat,validValue,roundToNearest, ignoredKey,isReserved} from './validateFunctions.js'
+
+//keyValValidator takes as a parameter cmcdJson, which is a javascript object. The function iterates through it validating every key value pair.
+export function keyValValidator(cmcdJson,errors){
+    for(var key in cmcdJson){
+        isReserved(errors,key);
+        if((keyTypes[key]==='Integer') && (cmcdJson[key]<0)){
+            console.log('The \''+ key +'\' value must greater than 0.');
+            errors.push(createError('invalid-value',key,cmcdJson[key]));
+        }
+        switch(key){
+            case 'bl':
+                roundToNearest(errors,key,cmcdJson[key],100,'ms');
+                if(!('ot' in cmcdJson)){
+                    console.log('The \'' + key + '\'key should only be sent with the \'ot\' key.');
+                    errors.push(createError('invalid-value',key,cmcdJson[key]));
+                }
+                break;
+            case 'cid':
+                maxLength(errors,key,cmcdJson[key]);
```

Comment:
since the cmcdjson[key] is common to all validator functions, you can create a const with its value before the switch case.
const keyValue = cmcdjson[key]